Newer `scipy` version triggering `UserWarning`

The newest version of `scipy` contains a new optimization shortcut for the case of lower and upper bounds being equal. This procedure passes a non-writable array without copying it. As `botorch` calls `torch.from_numpy` on this array, a `UserWarning` is triggered. This causes our current tests to fail, see #843 

Proposed solution:
- [ ] Potentially silence the error temporarily?
- [ ] Just accept that issue for now as it only concerns a single test
- [x] Create PRs at `scipy`/`botorch` for fixes: Turns out `scipy` is sufficient
- [ ] Wait for fix on `scipy` being accepted and merged: https://github.com/scipy/scipy/pull/25541

Opinions on whether or not to temporarily fix @AdrianSosic @Scienfitz @kalama-ai ?

This only happens for `scipy>=1.18`. Full tests installed a lower version `1.17.1` while core tests installed `1.18.0` (please don't ask me why), but this is the reason for that.

This is currently trigger by one of the cardinality tests - probably because we create an edge case situation here by setting bounds to the same values (or `botorch` doing that internally), would need to dig deeper into that. This is the exact error:

```bash
FAILED tests/constraints/test_subset_constraints_hybrid.py::test_subset_constraints_hybrid[continuous_cardinality] - UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at /pytorch/torch/csrc/utils/tensor_numpy.cpp:213.)
```

--- a/pytest.ini
+++ b/pytest.ini
@@ -38,6 +38,8 @@ filterwarnings =
     ignore:Sparse invariant checks are implicitly disabled.*:UserWarning:linear_operator
     ; https://github.com/stanfordmlgroup/ngboost/issues/408
     ignore:.*friedman_mse.*:FutureWarning:sklearn.tree._classes
+    ; https://github.com/scipy/scipy/pull/25541
+    ignore:.*NumPy array is not writable.*:UserWarning:botorch.generation.gen
 
     ; Numerics/optimization
     ; ---------------------

